lib/trycourier/exceptions.rb

```diff
@@ -0,0 +1,37 @@
+require "trycourier/version"
+require "net/http"
+require "json"
+require "openssl"
+
+module Courier
+  #ResponseError in order to maintain v1.0.2 signature.
+  class InputError < StandardError; end
+  class ResponseError < StandardError; end
+  class CourierAPIError < ResponseError; end  
+
+  class ErrorHandler
+    def self.check_err(res)
+      code = res.code.to_i
+      obj = JSON.parse res.read_body
+
+      if code < 400
```

Comment:
This would run this code receiving a 3xx response. I highly doubt an API would ever redirect you, I mean it could, but what sane one would ever? lol. I would check for explicit status codes here, but its not a blocker.